File: java/L200/LeetCode0219.java
package L200;

import java.util.HashMap;

public class LeetCode0219 {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        if (k == 0) {
            return false;
        }
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0;i < nums.length;++i) {
            if (map.containsKey(nums[i]) && i - map.get(nums[i]) <= k) {
                return true;
            }
            map.put(nums[i], i);
        }
        return false;
    }
}

File: cpp/L200/LeetCode0219.cc
#include <unordered_map>
#include <vector>
using std::vector;

class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        std::unordered_map<int, int> mp;
        for (int i = 0;i < nums.size();++i) {
            auto it{mp.find(nums[i])};
            if (it != mp.end() && i - it->second <= k) {
                return true;
            } else {
                mp[nums[i]] = i;
            }
        }      
        return false;
    }
};